bigchaindb/common/transaction.py

```diff
@@ -125,11 +118,66 @@ def from_dict(cls, data):
             except TypeError:
                 # NOTE: See comment about this special case in
                 #       `Input.to_dict`
-                fulfillment = Fulfillment.from_dict(data['fulfillment'])
+                fulfillment = _fulfillment_from_details(data['fulfillment'])
         fulfills = TransactionLink.from_dict(data['fulfills'])
         return cls(fulfillment, data['owners_before'], fulfills)
 
 
+def _fulfillment_to_details(fulfillment):
+    """
+    Encode a fulfillment as a details dictionary
+
+    Args:
+        fulfillment: Crypto-conditions Fulfillment object
+    """
+
+    if fulfillment.type_name == 'ed25519-sha-256':
+        return {
+            'type': 'ed25519-sha-256',
+            'public_key': base58.b58encode(fulfillment.public_key),
+        }
+
+    if fulfillment.type_name == 'threshold-sha-256':
+        subfulfillments = [
+            _fulfillment_to_details(cond['body'])
+            for cond in fulfillment.subconditions
+        ]
+        return {
+            'type': 'threshold-sha-256',
+            'threshold': fulfillment.threshold,
+            'subfulfillments': subfulfillments,
+        }
+
+    raise UnsupportedTypeError(fulfillment.type_name)
+
+
+def _fulfillment_from_details(data, limit=10, depth=0):
```

Comment:
I'd rename this method to something that empathizes even more that we're taking a jsonized condition and loading it into Python memory as a "**condition** object"